python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_defs.py

```diff
@@ -0,0 +1,101 @@
+import re
+from typing import Any, Dict, List, Optional, Union
+
+from dagster import (
+    AssetExecutionContext,
+    AssetsDefinition,
+    AssetSpec,
+    MaterializeResult,
+    multi_asset,
+)
+from dagster._annotations import experimental
+
+from dagster_embedded_elt.sling.resources import SlingMode, SlingResource
+
+
+@experimental
+def build_sling_asset(
+    asset_spec: AssetSpec,
+    source_stream: str,
+    target_object: str,
+    mode: SlingMode = SlingMode.FULL_REFRESH,
+    primary_key: Optional[Union[str, List[str]]] = None,
+    update_key: Optional[Union[str, List[str]]] = None,
+    source_options: Optional[Dict[str, Any]] = None,
+    target_options: Optional[Dict[str, Any]] = None,
+    sling_resource_key: str = "sling",
+) -> AssetsDefinition:
+    """Asset Factory for using Sling to sync data from a source stream to a target object.
+
+    Args:
+        asset_spec (AssetSpec): The AssetSpec to use to materialize this asset.
+        source_stream (str): The source stream to sync from. This can be a table, a query, or a path.
+        target_object (str): The target object to sync to. This can be a table, or a path.
+        mode (SlingMode, optional): The sync mode to use when syncing. Defaults to SlingMode.FULL_REFRESH.
+        primary_key (Optional[Union[str, List[str]]], optional): The optional primary key to use when syncing.
+        update_key (Optional[Union[str, List[str]]], optional): The optional update key to use when syncing.
```

Comment:
It would be nice to clarify that this is not an asset key. Threw me for a second.